File: library/src/main/java/cn/wandersnail/widget/listview/BaseViewHolder.java
package cn.wandersnail.widget.listview;

import android.view.View;

import androidx.annotation.NonNull;

/**
 * date: 2019/8/22 18:57
 * author: zengfansheng
 */
public interface BaseViewHolder<T> {
    void onBind(@NonNull T item, int position);
    
    @NonNull
    View createView();
}

File: library/src/main/java/com/snail/widget/listview/BaseViewHolder.kt
package com.snail.widget.listview

import android.view.View

/**
 * 主要用于ListView的item布局创建及数据设置
 */
abstract class BaseViewHolder<T> {
    /**
     * 和Adapter绑定了，可在此设置View的数据，更新View
     */
    abstract fun onBind(item: T, position: Int)

    /**
     * 创建界面
     */
    abstract fun createView(): View
}